swap/src/protocol/bob/event_loop.rs

```diff
@@ -164,22 +165,26 @@ impl EventLoop {
                         _ => {}
                     }
                 },
-                Some((request, responder)) = self.spot_price_requests.next().fuse() => {
+
+                // Handle to-be-sent requests for all our network protocols.
+                // Use `self.is_connected_to_alice` as a guard to "buffer" requests until we are connected.
+                Some((request, responder)) = self.spot_price_requests.next().fuse(), if self.is_connected_to_alice() => {
```

Comment:
❓ Does the spot price request have a timeout? Otherwise the program might hang upon startup if Alice is not available.

Reply:
Yes, all request-response protocols have the default timeout of 20 seconds.